File: VB/WindowsApplication3/MVVM/Data/DataAccess.vb
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Linq
Imports MVVMSample.MVVM.Data

Namespace MVVMSample
    Public Class DataAccessHelper
        Public Shared Function GetCustomers() As BindingList(Of Customer)
            Dim dt As New BindingList(Of Customer)()
            For i As Integer = 0 To 9
                dt.Add(New Customer With { _
                    .ID = i, _
                    .FirstName = String.Format("Name {0}", i), _
                    .LastName = String.Format("Last Name {0}", i) _
                })
            Next i
            Return dt
        End Function
        Public Shared Function GetOrders() As BindingList(Of Order)
            Dim dt As New BindingList(Of Order)()
            For i As Integer = 0 To 9
                dt.Add(New Order With { _
                    .ID = i, _
                    .OrderDate = Date.Today.AddDays(-i), _
                    .Freight = i * 100 + 100 _
                })
            Next i
            Return dt
        End Function
    End Class
End Namespace

File: CS/WindowsApplication3/MVVM/Data/DataAccess.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Linq;
using MVVMSample.MVVM.Data;

namespace MVVMSample {
    public class DataAccessHelper {
        public static BindingList<Customer> GetCustomers() {
            BindingList<Customer> dt = new BindingList<Customer>();
            for (int i = 0; i < 10; i++)
            {
                dt.Add(new Customer { ID = i, FirstName = string.Format("Name {0}", i), LastName = string.Format("Last Name {0}", i) });
            }
            return dt;
        }
        public static BindingList<Order> GetOrders() {
            BindingList<Order> dt = new BindingList<Order>();
            for (int i = 0; i < 10; i++) {
                dt.Add(new Order { ID = i, OrderDate = DateTime.Today.AddDays(-i), Freight = i * 100 + 100 });
            }
            return dt;
        }
    }
}
